modules/javafx.graphics/src/main/java/com/sun/javafx/tk/Toolkit.java

```diff
@@ -703,6 +703,15 @@ protected static final double clampStopOffset(double offset) {
                            float dashOffset);
 
     public abstract int getKeyCodeForChar(String character);
+    /**
+     * The default implementation bridges into the existing getKeyCodeForChar call.
+     */
+    public boolean getKeyCanGenerateCharacter(KeyEvent event, String character) {
```

Comment:
I forgot the `KeyEvent` is needed to extract the hardware code. Depending on whether this hardware code remains a hidden variable this may be for the best (or can pass both `KeyCode` and the hardware code).

Reply:
I'll change the call over to `canKeyGenerateCharacter`, the longer form was starting to annoy me too.